libr/util/hex.c

```diff
@@ -450,6 +450,31 @@ R_API int r_hex_str2bin(const char *in, ut8 *out) {
 	return nibbles / 2;
 }
 
+// get the hex chars from start of string, until first non-hex char, as a heap
+// allocated ut8* buffer
+R_API int r_hex_str2bin_until_new(const char *in, ut8 **out) {
+	r_return_val_if_fail (in && out && !*out, -1);
+	size_t len = strlen (in);
+	if (len <= 1) {
+		return 0;
+	}
+	len = (len + 1) / 2;
+
+	ut8 *buf = NULL;
+	size_t nibbles = 0;
+	if ((buf = malloc (len)) != NULL) {
```

Comment:
dont assign in if statements and != NULL is implicit

Reply:
you can define and assign in the same line `ut8 *buf = mallloc(len); if (!buf) {...}`